ln-gateway/src/ln.rs

```diff
@@ -52,7 +53,8 @@ impl LnRpc for Mutex<cln_rpc::ClnRpc> {
         match pay_result {
             Ok(cln_rpc::Response::Pay(pay_success)) => {
                 debug!("Successfully paid invoice");
-                Ok(pay_success.payment_preimage.to_vec().try_into().unwrap())
+                let pv: [u8; 32] = pay_success.payment_preimage.to_vec().try_into().unwrap();
+                Ok(Preimage(pv))
```

Comment:
This is very ugly but necessary (could make `unwrap` an `expect`, but nothing more we can do). It would be nice if you could try to upstream a change to [`cln_rpc`](https://github.com/ElementsProject/lightning/tree/master/cln-rpc) that implements `From<Secret> for [u8;32]`. That way we could get rid of the heap allocation here.

Reply:
Patch to cln got merged :) https://github.com/ElementsProject/lightning/pull/5642